std/algorithm/iteration.d

```diff
@@ -3855,46 +3864,127 @@ if (is(typeof(binaryFun!pred(r.front, s)) : bool)
     return Result(r, s);
 }
 
-///
+/// Basic splitting with characters and numbers.
 @safe unittest
 {
     import std.algorithm.comparison : equal;
-    import std.ascii : toLower;
-    import std.range : empty, retro;
 
-    // Basic splitting with characters and numbers.
-    assert(equal(splitter("a|bc|def", '|'), [ "a", "bc", "def" ]));
+    assert("a|bc|def".splitter('|').equal([ "a", "bc", "def" ]));
 
     int[] a = [1, 0, 2, 3, 0, 4, 5, 6];
     int[][] w = [ [1], [2, 3], [4, 5, 6] ];
-    assert(equal(splitter(a, 0), w));
+    assert(a.splitter(0).equal(w));
+}
 
-    // Adjacent separators.
-    assert(equal(splitter("a|b||c", '|'), [ "a", "b", "", "c" ]));
-    assert(equal(splitter("hello  world", ' '), [ "hello", "", "world" ]));
+/// Adjacent separators.
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
 
-    a = [ 1, 2, 0, 0, 3, 0, 4, 5, 0 ];
-    w = [ [1, 2], [], [3], [4, 5], [] ];
-    assert(equal(splitter(a, 0), w));
+    assert("|ab|".splitter('|').equal([ "", "ab", "" ]));
+    assert("ab".splitter('|').equal([ "ab" ]));
+
+    assert("a|b||c".splitter('|').equal([ "a", "b", "", "c" ]));
+    assert("hello  world".splitter(' ').equal([ "hello", "", "world" ]));
+
+    auto a = [ 1, 2, 0, 0, 3, 0, 4, 5, 0 ];
+    auto w = [ [1, 2], [], [3], [4, 5], [] ];
+    assert(a.splitter(0).equal(w));
+}
+
+/// Use a sub-range for splitting
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
+
+    assert("a=>bc=>def".splitter("=>").equal([ "a", "bc", "def" ]));
+    assert("a|b||c".splitter("||").equal([ "a|b", "c" ]));
+    assert("hello  world".splitter("  ").equal([ "hello", "world" ]));
 
-    // Empty and separator-only ranges.
-    assert(splitter("", '|').empty);
-    assert(equal(splitter("|", '|'), [ "", "" ]));
-    assert(equal(splitter("||", '|'), [ "", "", "" ]));
+    int[] a = [ 1, 2, 0, 0, 3, 0, 4, 5, 0 ];
+    int[][] w = [ [1, 2], [3, 0, 4, 5, 0] ];
+    assert(a.splitter([0, 0]).equal(w));
 
-    // Leading separators, trailing separators, or no separators.
-    assert(equal(splitter("|ab|", '|'), [ "", "ab", "" ]));
-    assert(equal(splitter("ab", '|'), [ "ab" ]));
+    a = [ 0, 0 ];
+    assert(a.splitter([0, 0]).equal([ (int[]).init, (int[]).init ]));
+
+    a = [ 0, 0, 1 ];
+    assert(a.splitter([0, 0]).equal([ [], [1] ]));
+}
 
-    // Predicate functions.
-    assert(equal(splitter!"a.toLower == b"("abXcdxef", 'x'),
+/// Custom predicate functions.
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
+    import std.ascii : toLower;
+
+    assert("abXcdxef".splitter!"a.toLower == b"('x').equal(
```

Comment:
If you're changing the code, it'd be a good time to update this to a lambda. Same throughout

Reply:
Agreed, but it's not trivial (frame access issues). Hence, I moved this into a separate issue https://issues.dlang.org/show_bug.cgi?id=18470